### PR TITLE
Remove dependency on wheel < 0.32.0

### PR DESCRIPTION
**Pull request by sublinus**
This PR enables use of `wheel>=0.32.0` with devpi-builder.

That is achieved through the use of `wheel-filename` and `wheel-inspect` as a replacement for wheel internal api calls, which stopped working for wheel versions >0.32.0 (#107 ).

Python 2 support is dropped here due to incompatibility with the newly introduced libraries.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,20 @@ Changelog
 This lists the most important changes for each release.
 
 
+Version 4.1.0 – UNRELEASED
+==========================
```
Comment (theMarix) [inline CHANGELOG.rst:9]: For consistency with previous versioning this should be 5.0.0.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,20 @@ Changelog
 This lists the most important changes for each release.
 
 
+Version 5.0.0 – UNRELEASED
+==========================
+
+Fixed
+-----
+
+* Reliance on the internal wheel api, which blocked use of latest wheel versions is now resolved.
```
Comment (Fra-nk) [inline CHANGELOG.rst:14]: Nitpick, but I prefer an imperative mood here, e.g.
```suggestion
* Remove dependency on the internal ``wheel`` API.
This blocked the use of ``wheel`` versions > 0.32. 
```

File: devpi_builder/wheeler.py [hunk 3]
```diff
@@ -10,9 +10,10 @@
 import subprocess
 import tempfile
 
-from wheel.install import WheelFile, BadWheelFile
-from wheel.util import matches_requirement
-
+from packaging import tags
+from pkg_resources import Requirement, Distribution
+from wheel_filename import parse_wheel_filename, InvalidFilenameError
+from wheel_inspect import inspect_wheel, Wheel
```
Comment (Fra-nk) [inline devpi_builder/wheeler.py:16]: PEP8 nitpick:
```suggestion
from pkg_resources import Distribution, Requirement

from wheel_filename import InvalidFilenameError, parse_wheel_filename
from wheel_inspect import Wheel, inspect_wheel
```

File: devpi_builder/wheeler.py [hunk 4]
```diff
@@ -38,12 +39,29 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         shutil.rmtree(self.scratch_dir)
 
+    def _matches_requirement(self, requirement, wheels):
+        """
+        List of wheels matching a requirement.
```
Comment (Fra-nk) [inline devpi_builder/wheeler.py:44]:
```suggestion
        List wheels matching a requirement.
        
```

File: devpi_builder/wheeler.py [hunk 5]
```diff
@@ -79,17 +97,27 @@ def is_pure(wheel):
...
+            .get("wheel", {})
+            .get("root_is_purelib")
+    )
 
 
 def is_compatible(package):
     """
     Check whether the given python package is a wheel compatible with the
     current platform and python interpreter.
+    Compatibility is based on PEP425
```
Comment (Fra-nk) [inline devpi_builder/wheeler.py:112]:
```suggestion

    Compatibility is based on PEP425
```

File: requirements.in [hunk 6]
```diff
@@ -1,9 +1,11 @@
--e .
```
Comment (Fra-nk) [inline requirements.in:1]: Why are you abandoning pulling in the `setup.py` requirements?
Doesn't this duplicate information between here and `setup.py`?

File: .travis.yml [hunk 7]
```diff
@@ -1,8 +1,8 @@
 language: python
 python:
-- '2.7'
 - '3.5'
 - '3.6'
+- '3.7'
```
Comment (Fra-nk) [inline .travis.yml:5]: Python 3.8 is "official" already.
Should we just add it here?

Comment (sublinus) [reply .travis.yml:5]: That would make sense in my opinion. Well spotted.

File: requirements.in (hunk 6 above)
Comment (sublinus) [reply requirements.in:1]: Yes, you're right this duplicates information. Will fix this.

File: .travis.yml (hunk 7 above)
Comment (sublinus) [reply .travis.yml:5]: There are multiple issues with dependencies on Python 3.8 (see [travis](https://travis-ci.org/github/blue-yonder/devpi-builder/jobs/672443191)).
I would propose moving forward with this PR and looking into the issues connected to Python 3.8 in another PR.

**Comment by Fra-nk** [comment]
Thanks, LGTM.

File: .travis.yml (hunk 7 above)
Comment (Fra-nk) [reply .travis.yml:5]: So much for my statement regarding surprises :D ...
Well, thanks for trying :D...

**Comment by sublinus** [comment]
I repinned the requirements again in order to validate that tests also run fine with newer wheel versions and now have the tests working on Python 3.8, but failing for 3.5.
Thus I would drop support for Python 3.5 and lower and introduce it for Python 3.7 + 3.8.

**Comment by Fra-nk** [comment]
With all the important `async(io)` changes in `3.6` and `3.7` I assume, _most_ people who migrate to Py3 won't bother with `3.5`, so this sounds good to me. :)

File: CHANGELOG.rst [hunk 8]
```diff
@@ -5,6 +5,21 @@ Changelog
...
+-----
+
+* Remove dependency on the internal ``wheel`` API.
+  This blocked the use of ``wheel`` versions >= 0.32.
+
+Removed
+-------
+
+* Python 2 is no longer supported.
+
```
Comment (Fra-nk) [inline CHANGELOG.rst:21]:
```suggestion
* Python 3.5 is no longer supported.

```

**Pull request merged**